pkg/build/generator/generator.go

```diff
@@ -199,6 +199,10 @@ func (g *BuildGenerator) Instantiate(ctx kapi.Context, request *buildapi.BuildRe
 		return nil, err
 	}
 
+	if _, ok := bc.Annotations[buildapi.BuildConfigDeletionAnnotation]; ok {
+		return nil, fmt.Errorf("can't instantiate from BuildConfig %s/%s: BuildConfig is being deleted", bc.Namespace, bc.Name)
+	}
```

Comment:
Is this error going to get retried by the retry controller?  we want to make sure this is treated as a FatalError, otherwise we're going to pointlessly retry it.

see:
https://github.com/openshift/origin/blob/master/pkg/build/controller/factory/factory.go#L320-L331
https://github.com/openshift/origin/blob/master/pkg/build/controller/factory/factory.go#L159-L178